### PR TITLE
Add a proper `File` class

### PR DESCRIPTION
**Pull request by flo12356**
### Description

Add class `File` as part of the wrapper for littlefs.

Fixes parts of #273
Fixes parts of #186
Fixes parts of #209

### CODE REVIEW

File: Sts1CobcSw/FileSystem/LfsWrapper.ipp [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <Sts1CobcSw/FileSystem/LfsWrapper.hpp>
+
+#include "Sts1CobcSw/FileSystem/ErrorsAndResult.hpp"
```
Comment (PatrickKa) [inline Sts1CobcSw/FileSystem/LfsWrapper.ipp:5]: Only the corresponding .hpp file may be included in an .ipp file, so this line must be removed.

File: Sts1CobcSw/Outcome/CMakeLists.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-target_link_libraries(Sts1CobcSw_Outcome INTERFACE rodos::rodos)
+# target_link_libraries(Sts1CobcSw_Outcome INTERFACE rodos::rodos)
```
Comment (PatrickKa) [inline Sts1CobcSw/Outcome/CMakeLists.txt:1]: I guess this is related to the whole `USE_NO_RODOS` issue. We'll probably need two `Outcome` targets: one with and one without Rodos. The unit tests can then link to the one without Rodos while the rest uses the "normal" version.

**Review by PatrickKa — CHANGES_REQUESTED** [review]
- `add_compile_definitions(USE_NO_RODOS)` in the top-level CML file seems wrong
- `Read/Write()` should be like `fram::ReadFrom/WriteTo()`, i.e. they should take byte spans:

  ~~~c++
  template<std::size_t extent>
  [[nodiscard]] auto Read(std::span<Byte, extent> data) const -> Result<int>;
  template<std::size_t extent>
  [[nodiscard]] auto Write(std::span<const Byte, extent> data) -> Result<int>;
  ~~~

- Rename `Read/WriteInternal()` to just `Read/Write()`. The private functions have different parameters, so there is no reason to give them different names for disambiguation (function overloading FTW).
- Use the same order for the public and private `Read/Write()` functions

File: Sts1CobcSw/FileSystem/LfsWrapper.ipp (hunk 1 above)
Comment (flo12356) [reply Sts1CobcSw/FileSystem/LfsWrapper.ipp]: Fixed in commit https://github.com/SpaceTeam/STS1_COBC_SW/pull/287/commits/e6efcca7d93be7354637641ebe498b194f0e4f44

File: Sts1CobcSw/Outcome/CMakeLists.txt (hunk 2 above)
Comment (flo12356) [reply Sts1CobcSw/Outcome/CMakeLists.txt]: Readded in https://github.com/SpaceTeam/STS1_COBC_SW/pull/287/commits/b395f28ac8899dddc896183bcb19de1f1184792b

**Review by PatrickKa — CHANGES_REQUESTED** [review]
- Make `CreateLockFile()` and `Close()` `const`
- In `Open()` when calling `CreateLockFile()` either use `OUTCOME_TRY(file.CreateLockFile());` or create (and return) a new, specific error code for this, like `fileLocked`
- Instead of removing the lock file in `MoveConstructFrom()` by calling `Close()` and creating it again later, we should add a private `CloseAndKeepLockFile()` function that we call instead. `Close()` should then also call this function and additionally remove the lock file.

**Review by PatrickKa — APPROVED** [review]

**Pull request merged**